src/solver/conv/conv_wino_fury_RxS.cpp

```diff
@@ -0,0 +1,386 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/solver.hpp>
+
+#include <miopen/conv/invokers/gcn_asm_wino.hpp>
+#include <miopen/conv/kernel_interface/winograd_kernel_interface.hpp>
+#include <miopen/env.hpp>
+#if !MIOPEN_USE_COMGR
+#include <miopen/kernel_build_params.hpp>
+#endif
+#include <miopen/stringutils.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_AMD_WINOGRAD_FURY_RXS_F2X3)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_AMD_WINOGRAD_FURY_RXS_F3X2)
+
+#define IS2X3 (Winodata == 2 && Winofilter == 3)
+#define IS3X2 (Winodata == 3 && Winofilter == 2)
+
+constexpr std::size_t sync_buffer_size = 2048;
+
+namespace miopen {
+namespace solver {
+namespace conv {
+
+using ProblemDescription            = miopen::conv::ProblemDescription;
+using WinoShaderArgsV40             = miopen::conv::WinoShaderArgsV40;
+using WinoShaderActivationModeV40_t = miopen::conv::WinoShaderActivationModeV40_t;
+using WinoShaderFlagsV40            = miopen::conv::WinoShaderFlagsV40;
+
+namespace {
+
+constexpr uint32_t PowOf2(uint32_t exp) { return 1U << exp; }
```

Comment:
Shifting more or equal to the number of bits is not well defined, probably we need some sort of `static_assert` here.

Reply:
This function is used only in this file, besides, I noticed that the compiler perfectly catches such situations

Reply:
@averinevg Does it issue a warning or error?

Reply:
@atamazov  @CAHEK7  It issues an error, but for constexpr variables only. However, I don't think this could be a problem here.

Reply:
@averinevg [Recommendation]
```C++
if (exp >= sizeof(1U)*8) throw std::logic_error("UB: Too big exp");
```
In case of UB:
- if return value of function is `constexpr` then you'll get build error
- otherwise exception will happen at run time.